Dockerfile

```
@@ -1,41 +1,113 @@
+# 1. Build Dependencies Stage with enhanced caching
+###############################################################################
+FROM elixir:1.18.3-otp-27-slim AS deps
 
 WORKDIR /app
 
-# Setup Elixir tools and dependencies
+# Set Mix environment
+ENV MIX_ENV=prod
+
+# Install build tools
+RUN apt-get update \
```

Hadolint (2.12.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)